### ISSUE TITLE
Resource request: What goes into a GitHub repo?

### ISSUE THREAD

**Issue by nzufelt** [nzufelt/open_source_movement_csc630#5]
There are a lot of files that go into a GitHub repo which _aren't_ the code itself, such as a `README.md`, perhaps a `CONTRIBUTING.md` or a `CODE_OF_CONDUCT.md`, or others.  We should make a page of short descriptions of each of these and what they mean, perhaps with links to longer articles or blog posts about the purpose of each.  This would be especially helpful for those new to open source!

**Comment by ruddfawcett** [comment]
Definitely... Brainstorming here...

- `.gitignore`
    - A file that tells `git` which files to ignore when executing any sort of `git add` (relative or universal — using `git add .`
- `README.md`
    - A file that contains basic information about the repository, such as title and description, creator, links to resources for the project, a reference to the license of the project, and perhaps a few usage examples. Some `README.md` files list contributors to the projects; others don't.
- `CONTRIBUTING.md`
    - A file that GitHub will link to at the top of every new pull request. Outlines coding conventions in the project, or specifies other contributing guidelines, such as "must include unit tests."
- `CODE_OF_CONDUCT.md`
    - A file which outlines the code of conduct for that project and its community on GitHub, usually defining expected behavior throughout issues, pull requests, or general discussion.
- `LICENSE`
    - A file which outlines the licensing for the project. Licenses specify the range of use and modification of (with and without attribution) any publically released code or graphics. A great website for comparing licenses is [TLDRLegal — Software Licenses Explained in Plain English](https://tldrlegal.com/).
- `CHANGELOG.md`
    - A file which outlines various changes in a projects lifetime. Usually just includes changes on a per-release basis, but may include more detailed changes. [Here is an example of a great changelog](https://github.com/facebook/react/blob/master/CHANGELOG.md),  from Facebook's JavaScript library for building user interfaces, React.

**Comment by milesmcc** [comment]
Another great resource for understanding licenses is available [here](https://www.toptal.com/open-source/developers-guide-to-open-source-licenses).

**Comment by ruddfawcett** [comment]
Closing now due to inactivity, but let's reopen if we want to revisit this! 🎉
